backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3AWTCanvas.java

```diff
@@ -0,0 +1,541 @@
+				Lwjgl3AWTCanvas.this.setDisplayMode(displayMode.width, displayMode.height);
+				return true;
+			}
+
+//			public boolean shouldRender () {
+//				synchronized (this) {
+//					boolean rq = requestRendering;
+//					requestRendering = false;
+//					return rq || isContinuous;
+//				}
+//			}
+		};
+
+		if (!config.disableAudio && Gdx.audio == null) audio = new OpenALLwjgl3Audio();
+		if (Gdx.files == null) files = new Lwjgl3Files();
+		if (Gdx.net == null) net = new Lwjgl3Net(config);
+		input = new Lwjgl3AWTInput(this);
+		setGlobals();
+	}
+
+	protected void setDisplayMode (int width, int height) {
+	}
+
+	protected void setTitle (String title) {
+	}
+
+	@Override
+	public ApplicationListener getApplicationListener () {
+		return listener;
+	}
+
+	public AWTGLCanvas getCanvas () {
+		return canvas;
+	}
+
+	@Override
+	public Audio getAudio () {
+		return Gdx.audio;
+	}
+
+	@Override
+	public Files getFiles () {
+		return files;
+	}
+
+	@Override
+	public Graphics getGraphics () {
+		return graphics;
+	}
+
+	@Override
+	public Input getInput () {
+		return input;
+	}
+
+	@Override
+	public Net getNet () {
+		return net;
+	}
+
+	@Override
+	public ApplicationType getType () {
+		return ApplicationType.Desktop;
+	}
+
+	@Override
+	public int getVersion () {
+		return 0;
+	}
+
+	void setGlobals () {
+		Gdx.graphics = graphics;
+		Gdx.gl30 = graphics.getGL30();
+		Gdx.gl20 = Gdx.gl30 != null ? Gdx.gl30 : graphics.getGL20();
+		Gdx.gl = Gdx.gl30 != null ? Gdx.gl30 : Gdx.gl20;
+
+		Gdx.app = this;
+		if (audio != null) Gdx.audio = audio;
+		if (files != null) Gdx.files = files;
+		if (net != null) Gdx.net = net;
+		Gdx.graphics = graphics;
+		Gdx.input = input;
+	}
+
+	void create () {
+		try {
+			setGlobals();
+			graphics.initiateGL();
+//			canvas.setVSyncEnabled(graphics.config.vSyncEnabled);
+			listener.create();
+			lastWidth = Math.max(1, graphics.getWidth());
+			lastHeight = Math.max(1, graphics.getHeight());
+			listener.resize(lastWidth, lastHeight);
+			start();
+		} catch (Throwable ex) {
+			stopped();
+			exception(ex);
+		}
+	}
+
+	void render (boolean shouldRender) {
+		if (!running) return;
+
+		setGlobals();
+		canvas.setCursor(cursor);
+
+		int width = Math.max(1, graphics.getWidth());
+		int height = Math.max(1, graphics.getHeight());
+		if (lastWidth != width || lastHeight != height) {
+			lastWidth = width;
+			lastHeight = height;
+			Gdx.gl.glViewport(0, 0, lastWidth, lastHeight);
+			resize(width, height);
+			listener.resize(width, height);
+			shouldRender = true;
+		}
+
+		if (executeRunnables()) shouldRender = true;
+
+		// If one of the runnables set running to false, for example after an exit().
+		if (!running) return;
+
+//		shouldRender |= graphics.shouldRender();
+		shouldRender=true;
+		input.processEvents();
+		if (audio != null) audio.update();
+
+		if (shouldRender) {
+			graphics.update();
+			listener.render();
+			canvas.swapBuffers();
+		}
+
+//		Display.sync(getFrameRate() * instanceCount);
```

Comment:
You should be able to do that with the `Sync` class in the lwjgl3 backend.

Reply:
I'll try that, thanks!

Reply:
Works (after I initialized GLFW)